### DISCUSSION TITLE
mounting directories question

### DISCUSSION THREAD

**Discussion by ghost** [mbround18/valheim-docker#185] [Q&A] [answered]
I have a question regarding the named volumes/ mounted directories. Why is the `./valheim/server:/home/steam/valheim` saved on the server? I suppose the `backups `and the `saves `folder contain enough information for the container to restart without data loss? I ask this question as this is by far the biggest folder and everything I mount/use a volume goes through my servers backup process and this makes the backup real big real fast.

Thanks in advance!

**Comment by CosmicHorrorDev** [comment, marked as answer]
How large are backups and saves? It really shouldn't be too big unless you have a weird mounting situation

**Reply by mbround18** [reply]
It should be about 30mb for each backup, on your mounted directories inside the container it is best to NOT put all the directories together as that would cause the backups to be massive. 



@LovecraftianHorror, i wonder if I should itemize what should go into the backups to prevent potential user error? or should I take a peak at the directory before its backed up and throw an error for invalid folder configuration?

**Reply by ghost** [reply]
the backups and saves folder are small, at the moment for me 6.5M in total. But I was wondering about the mount of `./valheim/server:/home/steam/valheim` . As this folder is for me now 1003M. And everything seems to work fine if I do not mount this folder and only mount the saves and backups folder. Even when removing and recreating the container. I was just wondering if this folder contains specific information that needs to be persisted, but as I use docker-compose to create the containers the environmental entries there would just recreate the server in the correct way

**Reply by CosmicHorrorDev** [reply]
It seems like weird mounting setups leading to ballooning backups is pretty common so it would be nice if we could _force_ correct behavior somehow. I'm not sure how obvious we can make backups failing (people may just expect it to work and then get surprised if it was failing the whole time).

I guess the best solution there would be to only backup a known set of files? The only thing that makes me hesitant would be mods expecting extra files to be backed up, but we already make no guarantees about behavior when using mods, so I think we're in the clear there.

I think all we need to backup is the `worlds` folder and the three files for admin, ban, and permit list. On mobile so can't check the actual names.

**Reply by mbround18** [reply]
> the backups and saves folder are small, at the moment for me 6.5M in total. But I was wondering about the mount of `./valheim/server:/home/steam/valheim` . As this folder is for me now 1003M. And everything seems to work fine if I do not mount this folder and only mount the saves and backups folder. Even when removing and recreating the container. I was just wondering if this folder contains specific information that needs to be persisted, but as I use docker-compose to create the containers the environmental entries there would just recreate the server in the correct way

You do not need to mount the `server` directory if you dont care that it redownloads the game each time the container is restarted. If you want to install mods tho, you will need to mount that directory.

**Reply by CosmicHorrorDev** [reply]
@SamLoys that folder contains the actual server installation. It's not _required_ to be mounted, but is good to have access too for troubleshooting

**Reply by ghost** [reply]
Okay thanks @LovecraftianHorror and @mbround18 !  That is what I was wondering, my question is answered now :)
